playwright-e2e/dsm/component/tables/onc-history-table.ts

```diff
@@ -0,0 +1,303 @@
+import {expect, Locator, Page} from '@playwright/test';
+import Table from 'dss/component/table';
+import DatePicker from '../date-picker';
+import TextArea from '../../../dss/component/textarea';
+import {
+  InputTypeEnum,
+  OncHistoryInputColumnsEnum,
+  OncHistorySelectRequestEnum
+} from '../tabs/enums/onc-history-input-columns-enum';
+import {OncHistoryInputs} from '../tabs/models/onc-history-inputs';
+import {
+  OncHistoryInputsMapValue,
+  OncHistoryInputsTypes
+} from '../tabs/interfaces/onc-history-inputs-types';
+import {waitForResponse} from '../../../utils/test-utils';
+import Select from '../../../dss/component/select';
+import TissueInformationPage from '../../pages/tissue-information-page/tissue-information-page';
+import Button from '../../../dss/component/button';
+import {FillDate} from '../../pages/tissue-information-page/interfaces/tissue-information-interfaces';
+import Input from '../../../dss/component/input';
+
+export default class OncHistoryTable extends Table {
+  private readonly tissueInformationPage = new TissueInformationPage(this.page);
+
+  constructor(protected readonly page: Page) {
+    super(page, {cssClassAttribute: '.table'});
+  }
+
+  public async openTissueInformationPage(index: number): Promise<TissueInformationPage> {
+    await this.tissueRequestPageButton(index).click();
+    return this.tissueInformationPage;
+  }
+
+  public async selectDeselectRow(index: number): Promise<void> {
+    const selectRowCheckbox = this.selectRowCheckbox(index);
+    await expect(selectRowCheckbox, 'Select row checkbox is not visible').toBeVisible();
+    await selectRowCheckbox.click();
+  }
```

Comment:
Function is little confusing to me. Is it deselect or select a checkbox?

Reply:
In fact, it has the capability to perform both actions. I hadn't initially implemented the functionality to check whether the checkbox is selected or not, but I now understand your perspective and agree with it. I will modify it into a select method that includes the necessary checks. Thank you for the suggestion.